pdns/dynhandler.cc

```diff
@@ -269,8 +269,8 @@ string DLNotifyHandler(const vector<string>&parts, Utility::pid_t ppid)
   UeberBackend B;
   if(parts.size()!=2)
     return "syntax: notify domain";
-  if(!::arg().mustDo("master"))
-      return "PowerDNS not configured as master";
+  if(!::arg().mustDo("master") && !::arg().mustDo("slave-renotify"))
+      return "PowerDNS not configured as master or slave with re-notifications";
```

Comment:
> We only notify 'all' if kind = 0 (master) so slave domains never get notified
I do not understand what you mean. Currently PDNS has following outgoing NOTIFY logic:
1. zone is MASTER, or
2. zone is SLAVE and slave-renotify is enabled

This patch does not change this behaviour. But currently "pdns_control notify" supports only scenario 1, not scenario 2. This patch allows "pdns_control notify" in the second scenario.

> And second, if slave=no should we send out slave renotifies when notifying all?
You are right, I could add a check for pdns_control for this